instrumentation/opentelemetry-instrumentation-wsgi/src/opentelemetry/instrumentation/wsgi/__init__.py

```diff
@@ -383,13 +434,24 @@ def __call__(self, environ, start_response):
             environ: A WSGI environment.
             start_response: The WSGI start_response callable.
         """
+        req_attrs = collect_request_attributes(environ)
+        active_requests_count_attrs = {}
+        for attr_key in _active_requests_count_attrs:
+            if req_attrs.get(attr_key) is not None:
+                active_requests_count_attrs[attr_key] = req_attrs[attr_key]
```

Comment:
I think we should be setting the same attribute keys for a given instrument regardless if the value is available or not. If the value is not available, we should have some default like empty string or 0.

This is what [Prometheus/OM expects](https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#metric:~:text=Metrics%20with%20the%20same%20name%20for%20a%20given%20MetricFamily%20SHOULD%20have%20the%20same%20set%20of%20label%20names%20in%20their%20LabelSet.)

Reply:
Good point. My original intention was to filter out the attribute keys which are not available at all. For example I couldn't find `NET_HOST_NAME` in the collected requested attributes. Should we be setting it 0 all the time or just don't set it?

Reply:
We should use a value that will never, under any circumstances be also a valid value. Maybe `None`?

Reply:
@srikanthccv maybe I misread the code but it looks like the intention here is to handle it dynamically if they are or aren't available? If the options are "setting it 0 all the time or just don't set it" i would say don't set it in that case 👍 I guess my point is we should keep the full attribute key set as if there was a strict type for it imo

> Maybe None?

None isn't allowed attribute value though? https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/README.md#:~:text=Attribute%20values%20of%20null%20are%20not%20valid%20and%20attempting%20to%20set%20a%20null%20value%20is%20undefined%20behavior

Reply:
Talked about this during 16th Jun SIG meeting. This requires some broader discussion because we setting zero-values might not be obvious to users and can't be generalised to non-existence of attribute value because there can be cases where zero-values are the legitimate input. And only setting the attributes that will always exist might be useful given the very little info from tag set. We agreed to get this merged and continue discussion for long term solution.